internal/common/rules/globalrulescache_test.go

```diff
@@ -13,6 +15,66 @@ func TestRegisterNewRule(t *testing.T) {
 	}
 }
 
+func TestPullRuleIntoPlay(t *testing.T) {
+	registerTestRule()
+	_ = PullRuleIntoPlay("Kinda Test Rule 2")
+	cases := []struct {
+		name string
+		rule string
+		want error
+	}{
+		{
+			name: "normal working",
+			rule: "Kinda Test Rule",
+			want: nil,
+		},
+		{
+			name: "unidentified rule name",
+			rule: "Unknown Rule",
+			want: errors.Errorf("Rule '%v' is not available in rules cache", "Unknown Rule"),
+		},
+		{
+			name: "Rule already in play",
+			rule: "Kinda Test Rule 2",
+			want: errors.Errorf("Rule '%v' is already in play", "Kinda Test Rule 2"),
+		},
+	}
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			got := PullRuleIntoPlay(tc.rule)
+			testutils.CompareTestErrors(tc.want, got, t)
+		})
+	}
+}
+
+func TestPullRuleOutOfPlay(t *testing.T) {
+	registerTestRule()
+	_ = PullRuleOutOfPlay("Kinda Test Rule 2")
+	_ = PullRuleIntoPlay("Kinda Test Rule")
```

Comment:
This unfortunately won't bode well when you want to run tests in parallel. It's not great practice to alter constants/outside variables in tests, so we usually use *Dependency Injection*

You can instead pass in the pointer to `Available Rules` to the function :).

Reply:
Otherwise, just reset the state with a `defer`, this is common practice, but again means no parallel tests possible.

```
origAvailableRules := AvailableRules
defer func() {
    AvailableRules = orgAvailableRules
}
// do whatever you want here
```

Reply:
Agreed and pushed to branch